### ISSUE TITLE
verbose dump

### ISSUE THREAD

**Issue by twicki** [MeteoSwiss-APN/dawn#180]
## Issue
the -fpass-verbose currently adds the multistages as keys to the stencil. For iteration purposes it would be better to have one dict mulitstages that holds them. Same for lower levels of the tree.

## Additional information
The discussion led to the conclusion that we should have everything that does not require ordering (fields, caches, etc...) in dictionaries and everything that has an order in arrays:
```
{
  "Stencil": {
    "Fields": {
      "Field1": {
        "name": "b"
      },
      "Field2": {
        "name": "a"
      }
    },
    "MSS": [
      "ms1",
      "ms0",
      "ms2"
    ]
  }
}
```

## Example
```
// RUN: %gtclang% %file% -fpass-verbose

#include "gridtools/clang_dsl.hpp"
using namespace gridtools::clang;

stencil test {
  storage a, b;
  Do {
    vertical_region(k_start, k_end) {
      // copy stencil
      a = b;
    }
  }
};
````

**Comment by cosunae** [comment]
the -fpass-verbose crashes on horizontal_advection_wwcon (because of stencil functions). 
No worth fixing this until we fix the design of the metadata and IIR

**Comment by cosunae** [comment]
The ordering problem of the fields is solved in #184 
But we also wanted that multistages are a collection (instead of dict)
We will do that after the redesign, #184 is only a bugfix
